src/Compilers/Core/Portable/Syntax/SyntaxTriviaListBuilder.cs

```diff
@@ -135,21 +137,21 @@ public SyntaxTriviaList ToList()
                     case 2:
                         return new SyntaxTriviaList(default(SyntaxToken),
                             InternalSyntax.SyntaxList.List(
-                            _nodes[0].UnderlyingNode,
-                            _nodes[1].UnderlyingNode), position: 0, index: 0);
+                            _nodes[0].UnderlyingNode!,
```

Comment:
should this use RequiredUnderlyingNode?

Reply:
or is this actually a situation where we just want the compiler to accept it, we don't want to assert about it in debug builds?

Reply:
The use of `UnderlyingNode!` is deliberate here. The rational in the Nullable Annotation Guidance document is roughly: 

1. Use `!` when the method already verifies a value is non-null but the compiler just can't follow along
1. Use `Debug.Assert`, or more formal methods, when the method doesn't do the verification.

In this case the `switch` statement effectively verifies that the repsective nodes are non-null. Essentially `_nodes` will have a non-null value for whatever the `_count` of the list is. Hence the method already verifies so use `!`.